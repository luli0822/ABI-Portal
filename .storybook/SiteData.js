// Use this file to mock SiteDataContext and to use across stories and unit tests
import pngImage from './static/png-example.png';
import textSVGImage from './static/page-hero-text.svg';
import svgImage from './static/svg-example.svg';

// Mock contentful siteData
export const SiteData = {
	pathMap: {
		'/en_CA': {
			page: "/",
			query:{
				lang: "en-CA", pageId: "page-id"
			} 
		},
	},
	linkedAssets: {
		// PNG Image asset link
		'asset-id-1': {
			sys: {
				type: 'Asset',
				id: 'asset-id-1'
			},
			field: {
				title: {
					'en-CA': 'image title text'
				},
				file: {
					'en-CA': {
						url: pngImage,
						details: {
							size: 150795,
							image: {
								width: 300,
								height: 300
							}
						},
						fileName: "png-example.png",
						contentType: "image/png"
					}
				}

			}
		},
		// SVG Image asset link
		'asset-id-2': {
			sys: {
				type: 'Asset',
				id: 'asset-id-2'
			},
			field: {
				title: {
					'en-CA': 'image title text'
				},
				file: {
					'en-CA': {
						url: svgImage,
						details: {
							size: 150795,
							image: {
								width: 300,
								height: 300
							}
						},
						fileName: "svg-example.svg",
						contentType: "image/svg+xml"
					}
				}

			}
		}
	},
	previewAssets: {
		'asset-id-1':{
			 url: 'data:image/png;base64,someUriData', meta: { width: 300, height: 130 } 
		},
		// svg image preview asset
		'asset-id-2':{
			url: 'data:image/svg+xml;base64,someUriData', meta: { width: 	500, height: 120 } 
	 }
	},
	// keyValues: {},
	lang: 'en-CA',
	fieldOrFallback: (field, lang = 'en-CA') => (
		field && field[lang] ? field[lang] : undefined
	)
};

// A method to build Link data structure
const linkDataMock = (type) => ({
	sys: {
		id: 'page-link-id',
		contentType: {
			sys: {
				type: 'Link',
				linkType: 'ContentType',
				id: `${type}`,
			},
		},
	},
	fields: {
		title: {
			'en-CA': 'Link Text',
		},
		accessibleLabel: {
			'en-CA': 'Link accessible Text',
		},
		url: type === 'externalLink' ? {'en-CA': 'https://loblawdigital.co'} : undefined,
		page: type === 'pageLink' ? {
			'en-CA': {
				sys: {
					type: 'Link',
					linkType: 'Entry',
					id: 'page-id',
				},
			},
		} : undefined
	},
})

// Use this for pageLink structure mock
export const pageLinkMock = linkDataMock('pageLink')

// Use this externalLink structure mock
export const externalLinkMock = linkDataMock('externalLink')

// A method to build Media/image data structure
const imageDataMock = (imageUrl, type) => ({
	sys: {
		type: "Asset",
		id: type === 'image/png' ? "asset-id-1" : "asset-id-2" ,
	},
	fields: {
		title: {
			'en-CA': 'Image Title Text'
		},
		description: {
			'en-CA': 'Image description Text'
		},
		file: {
			'en-CA': {
				url: imageUrl,
				details: {
					size: 150795, 
					image: {
						width: 300,
						height: 300
					}
				},
				fileName: "png-example.png",
				contentType: type
			}
		}
	}
})

// You can add the needed image mocks for your porject
// Use this for PNG image structure mock
export const imagePNGMock = imageDataMock(pngImage, 'image/png')

// Use this for SVG image structure mock
export const textSVGMock = imageDataMock(textSVGImage, 'image/svg+xml')

// Use this for SVG logo image structure mock
export const logoSVGMock = imageDataMock(svgImage, 'image/svg+xml')

// Mock useNavigation data
export const navData = {
	logo: {
		primary: logoSVGMock,
		primaryAccessibleText: 'Accessible description of logo',
		alt: textSVGMock,
		altAccessibleText: 'Accessible description of logo'
	},
	langSwitchLink: {
		href: '/some-href',
		asPath: '/some-path',
		text: 'change lang',
		accessibleText: 'Switch lang to EN',
	},
	skipNavText: 'Skip to main content',
	headerNav: [
		{
			type: 'link',
			link: pageLinkMock,
			text: 'Home Page',
			accessibleText: undefined
		},
		{
			type: 'link',
			link: externalLinkMock,
			text: 'External Link',
			accessibleText: undefined
		},

	],
	footerNav: [
		{
			type: 'group',
			text: 'Link group',
			accessibleText: undefined,
			items: [
				{
					type: 'link',
					link: pageLinkMock,
					text: 'link 1',
					accessibleText: undefined
				},
				{
					type: 'link',
					link: externalLinkMock,
					text: 'link 2',
					accessibleText: undefined
				},
			]
		}
	]
}