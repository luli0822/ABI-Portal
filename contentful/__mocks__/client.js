import fetchSiteData from './fetchSiteData.json';
// Refresh mock siteData will disclude about-us
module.exports = {
	getClient: () => {},
	fetchSiteData: () => fetchSiteData,
};
