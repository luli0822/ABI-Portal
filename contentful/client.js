const { createClient } = require('contentful');
const config = require('./config');

function getClient() {
	const { space, accessToken, host, environment } = config;
	return createClient({
		space,
		accessToken,
		host,
		environment,
		resolveLinks: false,
		removeUnresolved: false,
	});
}

async function fetchSiteData(siteSlug) {
	const client = getClient();

	const allData = await client
		.getEntries({
			content_type: 'site',
			'fields.slug': siteSlug,
			locale: '*',
			include: 10,
		})
		.catch(err => {
			throw Error(err);
		});

	const siteEntry = allData.items[0];
	const linkedEntries = allData.includes.Entry.reduce(
		(acc, cur) => ({ ...acc, [cur.sys.id]: cur }),
		{}
	);

	const linkedAssets = allData.includes.Asset.reduce(
		(acc, cur) => ({ ...acc, [cur.sys.id]: cur }),
		{}
	);

	return {
		siteEntry,
		linkedEntries,
		linkedAssets,
	};
}

module.exports = {
	getClient,
	fetchSiteData,
};
