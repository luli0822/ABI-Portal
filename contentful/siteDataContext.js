// siteDataContext just exports a context that can be shared across the app.

import { createContext } from 'react';

export default createContext();
