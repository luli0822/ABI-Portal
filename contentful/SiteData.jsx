import React, { useState, useEffect } from 'react';
import PropTypes from 'prop-types';
// eslint-disable-next-line import/no-unresolved
import staticSiteData from '../data/siteData.json';
// eslint-disable-next-line import/no-unresolved
import previewAssets from '../data/assets.json';
import { pathMap } from './path-map';
import { fetchSiteData } from './client';
import config from './config';
import { langPropType } from './prop-types';
import siteDataContext from './siteDataContext';
// import { siteDomains, isProd } from '../site-config';
/**
 * Builds the siteData object for the siteDataContext and allows refreshing the site data in preview mode
 * @param {*} props
 */
export default function SiteData(props) {
	const { children, lang, pageId } = props;

	// start out with the static site data store in json before the build
	const [siteData, setSiteData] = useState(staticSiteData);
	const [isRefreshing, setIsRefreshing] = useState(false);

	async function refresh() {
		setIsRefreshing(true);
		const newSiteData = await fetchSiteData(config.siteSlug);
		setSiteData(newSiteData);
		setIsRefreshing(false);
	}

	useEffect(() => {
		// Refresh when the page loads in the client in preview mode.
		// This allows us to not build the preview site every time a piece of content is saved.
		// Instead, we just get the latest content on site load.
		if (config.isPreview) {
			refresh();
		}
	}, []);

	const { siteEntry, linkedEntries, linkedAssets } = siteData;

	// make keyValues nicer to work with
	// TODO: the values here are still just the keyValue/keyLink/keyRichText entries.
	// maybe we can make it even simpler and just actually map key:value?
	const keyValues = siteEntry.fields.keyValues
		? siteEntry.fields.keyValues[config.fallbackLang].reduce((acc, cur) => {
				const valueEntry = siteData.linkedEntries[cur.sys.id];
				return { ...acc, [valueEntry.fields.slug[config.fallbackLang]]: valueEntry };
		  }, {})
		: {};

	function isLink(field) {
		return field && field.sys && field.sys.type === 'Link';
	}

	/**
	 * Extract the field value in either current language or fallback language
	 * Or, if a link, return the linked object.
	 * TODO: is this doing too much? should we know when we're getting a link and when we're not?
	 * @param {object} field
	 * @param {string} fieldLang
	 */
	function fieldOrFallback(field, fieldLang = lang) {
		if (isLink(field)) {
			if (field.sys.linkType === 'Entry') return linkedEntries[field.sys.id];
			if (field.sys.linkType === 'Asset') return linkedAssets[field.sys.id];
		}

		const fieldValue = field ? field[fieldLang] || field[config.fallbackLang] : undefined;

		if (isLink(fieldValue)) {
			if (fieldValue.sys.linkType === 'Entry') return linkedEntries[fieldValue.sys.id];
			if (fieldValue.sys.linkType === 'Asset') return linkedAssets[fieldValue.sys.id];
		}

		return fieldValue;
	}

	/**
	 * Map properties of a pageLink or externalLink object to strings for use by the component
	 * @param {object} link link object
	 */
	function mapLink(link) {
		return {
			type: 'link',
			link,
			text: fieldOrFallback(link.fields.title),
			accessibleText: fieldOrFallback(link.fields.accessibleText),
		};
	}

	/**
	 * Map properties of an asset object to strings for use by the component
	 * @param {string} fieldName name of the field
	 * @param {object} linkObj link reference object, will be resolved to the actual media object
	 */
	function mapMedia(fieldName, linkObj) {
		const link = fieldOrFallback(linkObj);

		if (!link) return {};

		return {
			[fieldName]: link,
			[`${fieldName}AccessibleText`]: fieldOrFallback(link.fields.description),
		};
	}

	/**
	 * Map properties of a linkGroup object to strings for use by the component
	 * @param {object} link link object
	 * @param {function} mapItems a function that will map group items' into nav fields
	 */
	function mapGroup(link, mapItems) {
		return {
			type: 'group',
			text: fieldOrFallback(link.fields.title),
			accessibleText: fieldOrFallback(link.fields.accessibleText),
			items: mapItems(link.fields.navigation, fieldOrFallback),
		};
	}

	/**
	 * Map properties of a Contentful nav object to a simpler, standard props object with content in the current locale.
	 * This object is passed to the component, providing content values.
	 * @param {object} navField The Contentful block object
	 */
	function mapNavFields(navField) {
		const navList = fieldOrFallback(navField);
		// Map + resolve standard fields
		if (!navList) return undefined;

		const items = navList.map(itemLink => {
			const item = fieldOrFallback(itemLink);
			switch (item.sys.contentType.sys.id) {
				case 'linkGroup':
					return mapGroup(item, fieldOrFallback, mapNavFields);
				case 'pageLink':
				case 'externalLink':
					return mapLink(item, fieldOrFallback);
				default:
					return {};
			}
		});

		return items;
	}

	/** Map lang Switch object */
	function mapLangSwitch() {
		const langToggleKeyValue = fieldOrFallback(siteEntry.fields.languageToggle);

		const text = langToggleKeyValue && fieldOrFallback(langToggleKeyValue.fields.value);

		const accessibleTextKeyValue = fieldOrFallback(siteEntry.fields.languageToggleAccessibleText);

		const accessibleText =
			accessibleTextKeyValue && fieldOrFallback(accessibleTextKeyValue.fields.value);

		const altLang = lang === 'en-CA' ? 'fr-CA' : 'en-CA';

		// If the pageId isn't part of the defined routes in pathMap (error page in this case), we will be build the language switch link
		// to use index page as destination route. (siteEntry.fields.indexPage[config.fallbackLang].sys.id is the index page id)
		// Note: This senario will happen when you don't have an error page setup in contentful,
		// otherwise, the error page path will use the page from the pathMap

		// Get the page based on the pageId is defined/exists
		const pagePath = Object.entries(pathMap).find(
			([path, options]) => options.query.pageId === pageId && options.query.lang === altLang
		);

		// Get the index page from path map if the pageId is undefined
		const indexPage = Object.entries(pathMap).find(
			([path, options]) =>
				options.query.pageId === siteEntry.fields.indexPage[config.fallbackLang].sys.id &&
				options.query.lang === altLang
		);

		// pluck the "path" from the page definition
		const [altHref] = pagePath || indexPage;

		// UNCOMMENT THE LINES BELLOW If you require switching domains for English and French on prod.
		// uncomment isProd and siteDomains import line at the top of this file.
		// and make "altHref" a let instead of const above
		// altHref = isProd
		// 	? `${siteDomains[altLang]}${altHref}`
		// 	: altHref;

		return {
			// we return an href instead of href and asPath because the language switcher is always an <a> tag
			href: altHref,
			text,
			accessibleText,
		};
	}

	function buildIndexPageLink() {
		const [asPath, { page, query }] = Object.entries(pathMap).find(
			([path, options]) =>
				options.query.pageId === siteEntry.fields.indexPage[config.fallbackLang].sys.id &&
				options.query.lang === lang
		);

		const href = `${page}?pageId=${query.pageId}&lang=${lang}`;

		return {
			href,
			asPath,
		};
	}

	// Navigation
	const primaryLogo = mapMedia('primary', siteEntry.fields.siteLogo);

	const altLogo = mapMedia('alt', siteEntry.fields.siteLogo);

	// Build the site logo link
	const indexPageLink = buildIndexPageLink();

	const headerNav = mapNavFields(siteEntry.fields.navigation) || [];

	const footerNav = mapNavFields(siteEntry.fields.footerNavigation) || [];

	const langSwitchLink = mapLangSwitch();

	// Set the key value name as you have it in Contentful
	const skipNavText =
		keyValues['skip-nav-text'] && fieldOrFallback(keyValues['skip-nav-text'].fields.value, lang);

	const siteContext = {
		siteEntry,
		linkedEntries,
		linkedAssets,
		keyValues,
		pathMap,
		lang,
		fieldOrFallback,
		previewAssets,
		headerNav,
		footerNav,
		logo: {
			...primaryLogo,
			...altLogo,
			...indexPageLink,
		},
		indexPageLink,
		langSwitchLink,
		skipNavText,
	};

	const buttonStyles = {
		boxShadow: '3px 3px 5px 0px #AFB1BC',
		display: 'inline-block',
		backgroundColor: '#ffffff',
		color: '#000',
		border: '1px solid rgb(216, 216, 216)',
		borderRadius: '4px',
		padding: '2px 7px',
		textDecoration: 'none',
		fontSize: '12px',
		fontFamily: 'sans-Serif',
	};

	return (
		<siteDataContext.Provider value={siteContext}>
			{config.isPreview && (
				<div
					style={{
						position: 'fixed',
						top: '5px',
						left: '5px',
						zIndex: '2000',
					}}
					role="complementary"
				>
					<button type="button" onClick={refresh} style={buttonStyles}>
						{isRefreshing ? '...' : 'Refresh'}
					</button>
					<a
						href="/storybook"
						style={{
							...buttonStyles,
						}}
					>
						Component Library
					</a>
				</div>
			)}
			{children}
		</siteDataContext.Provider>
	);
}

SiteData.propTypes = {
	children: PropTypes.node.isRequired,
	lang: langPropType.isRequired,
	pageId: PropTypes.string,
};

SiteData.defaultProps = {
	pageId: undefined,
};
