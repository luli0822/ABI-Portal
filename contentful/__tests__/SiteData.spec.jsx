/* eslint-disable no-unused-vars */
import React, { useContext } from 'react';
import { shallow, mount } from 'enzyme';
import { act } from 'react-dom/test-utils';
import siteDataContext from '../siteDataContext';
import refreshedFetchedData from '../__mocks__/refreshedFetchSiteData.json';
import mockSiteData from '../../data/__mocks__/siteData.json';
import mockAssets from '../../data/__mocks__/assets.json';

//  mock staticSiteData
jest.mock('../../data/siteData.json', () => mockSiteData, { virtual: true });
// mock previewAssets
jest.mock('../../data/assets.json', () => mockAssets, { virtual: true });
// mock pathMap
jest.mock('../path-map');
// mock config
jest.mock('../config');
// mock client (fetchSiteData)
jest.mock('../client');

// eslint-disable-next-line import/first
import client from '../client';

const originalFetchSiteData = client.fetchSiteData;

const mockFetchSiteData = jest.fn(originalFetchSiteData);

client.fetchSiteData = mockFetchSiteData;

// eslint-disable-next-line import/first
import SiteData from '../SiteData';

// jest.mock('react', () => {
// 	const ActualReact = require.requireActual('react');
// 	return {
// 		...ActualReact,
// 		useContext: () => ({
// 			fieldOrFallback: mockFieldOrFallback,
// 			lang: mockLang,
// 			previewAssets: mockPreviewAssets,
// 		}),
// 	};
// });
const stringifyObject = obj =>
	JSON.stringify(obj)
		.split('"')
		.join('');

// Mock child component
function MockChildComponent() {
	const {
		siteEntry,
		linkedEntries,
		linkedAssets,
		keyValues,
		pathMap,
		lang,
		fieldOrFallback,
		previewAssets,
		langSwitchLink,
	} = useContext(siteDataContext);

	return (
		<div>
			<h1>Some Mock Component</h1>
			<h2>siteEntry</h2>
			<div>{stringifyObject(siteEntry)}</div>
			<h2>linkedEntries</h2>
			<div>{stringifyObject(linkedEntries)}</div>
			<h2>linkedAssets</h2>
			<div>{stringifyObject(linkedAssets)}</div>
			<h2>keyValues</h2>
			<div>{stringifyObject(keyValues)}</div>
			<h2>PathMap</h2>
			<div>{stringifyObject(pathMap)}</div>
			<h2>lang</h2>
			<div>{stringifyObject(lang)}</div>
			<h2>previewAssets</h2>
			<div>{stringifyObject(previewAssets)}</div>
			<h2>Render Link using fieldOrFallback</h2>
			{stringifyObject(fieldOrFallback(siteEntry.fields.navigation[lang][0], lang))}
			<h2>langSwitchLink</h2>
			<div>{stringifyObject(langSwitchLink)}</div>
		</div>
	);
}

afterAll(() => {
	jest.clearAllMocks();
	mockFetchSiteData.mockReset();
});

describe('SiteData', () => {
	it('Provides correct values for siteData', async () => {
		// On inital render, the refresh functions
		let component;
		await act(async () => {
			component = await mount(
				// eslint-disable-next-line react/jsx-props-no-spreading
				<SiteData lang="en-CA" pageId="site-home-page">
					<MockChildComponent />
				</SiteData>
			);
		});
		expect(component.debug()).toMatchSnapshot();
	});

	it('Refresh button is rendered inPreview', async () => {
		let component;
		await act(async () => {
			component = await mount(
				// eslint-disable-next-line react/jsx-props-no-spreading
				<SiteData lang="en-CA" pageId="site-home-page">
					<MockChildComponent />
				</SiteData>
			);
		});
		expect(component.find('button[type="button"]')).toHaveLength(1);
	});

	it('Check refresh() is updating site content with a new siteData set', async () => {
		let component;
		await act(async () => {
			component = await mount(
				// eslint-disable-next-line react/jsx-props-no-spreading
				<SiteData lang="en-CA" pageId="site-home-page">
					{/* <MockChildComponent /> */}
					<MockChildComponent />
				</SiteData>
			);
		});

		// this string is present in the rendering of MockChildComponent when using the default siteData
		expect(component.text()).toMatch('indexPage:{en-CA:{sys:{id:site-home-page}}}');

		mockFetchSiteData.mockReturnValue(refreshedFetchedData);

		await act(async () => {
			await component
				.find('button')
				.first()
				.simulate('click');
		});

		component.update();

		// this string is present in rendering of MockChildComponment when using the Refreshed Fetched Data
		expect(component.text()).toMatch('indexPage:{en-CA:{sys:{id:REFRESHED-site-home-page}}}');
	});
});
