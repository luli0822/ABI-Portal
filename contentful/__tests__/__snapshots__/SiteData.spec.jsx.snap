// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SiteData Provides correct values for siteData 1`] = `
"<SiteData lang=\\"en-CA\\" pageId=\\"site-home-page\\">
  <div style={{...}} role=\\"complementary\\">
    <button type=\\"button\\" onClick={[Function: refresh]} style={{...}}>
      Refresh
    </button>
    <a href=\\"/storybook\\" style={{...}}>
      Component Library
    </a>
  </div>
  <MockChildComponent>
    <div>
      <h1>
        Some Mock Component
      </h1>
      <h2>
        siteEntry
      </h2>
      <div>
        {sys:{space:{sys:{id:spaceId}},id:siteId,contentType:{sys:{id:site,type:link}}},fields:{title:{en-CA:en title,fr-CA:fr title},slug:{en-CA:site-slug},indexPage:{en-CA:{sys:{id:site-home-page}}},pages:{en-CA:[{sys:{id:about-us-page}}]},keyValues:{en-CA:[{sys:{type:Link,linkType:Entry,id:lang-toggle-key-value}},{sys:{type:Link,linkType:Entry,id:rich-text-key-value}}]},navigation:{en-CA:[{sys:{type:Link,linkType:Entry,id:about-us-page-link}},{sys:{type:Link,linkType:Entry,id:some-external-link}}]},languageToggle:{en-CA:{sys:{type:Link,linkType:Entry,id:lang-toggle-key-value}}},languageToggleAccessibleText:{en-CA:{sys:{type:Link,linkType:Entry,id:lang-toggle-accessible-text-key-value}}}}}
      </div>
      <h2>
        linkedEntries
      </h2>
      <div>
        {site-home-page:{sys:{id:site-home-page,space:{},contentType:{id:page,type:Link,linkType:ContentType}},fields:{title:{en-CA:title en,fr-CA:title fr},description:{en-CA:description en,fr-CA:description en},openGraphTitle:{en-CA:openGraphTitle en,fr-CA:openGraphTitle en},openGraphDescription:{en-CA:openGraphDescription en,fr-CA:openGraphDescription en},slug:{en-CA:home-pg-en,fr-CA:home-pg-fr}}},about-us-page:{sys:{id:about-us-page,space:{},contentType:{id:page,type:Link,linkType:ContentType}},fields:{title:{en-CA:title en,fr-CA:title fr},description:{en-CA:description en,fr-CA:description en},openGraphTitle:{en-CA:openGraphTitle en,fr-CA:openGraphTitle en},openGraphDescription:{en-CA:openGraphDescription en,fr-CA:openGraphDescription en},slug:{en-CA:about-pg-en,fr-CA:about-pg-fr}}},about-us-page-link:{sys:{id:about-us-page-link,space:{},contentType:{sys:{type:Link,linkType:Entry,id:pageLink}}},fields:{title:{en-CA:page link title,fr-CA:page link title fr},page:{en-CA:{sys:{id:about-us-page},fields:{slug:{en-CA:some slug}}}},hash:{en-CA:s}}},some-external-link:{sys:{id:some-external-link,space:{},contentType:{sys:{id:externalLink,type:Link,linkType:Entry}}},fields:{title:{en-CA:External link text,fr-CA:External link text fr},url:{en-CA:https://test.com,fr-CA:https://test.com/fr}}},lang-toggle-key-value:{sys:{id:lang-toggle-key-value,space:{},contentType:{id:keyValue}},fields:{slug:{en-CA:lang-toggle-slug},value:{en-CA:toggle English,fr-CA:toggle French}}},lang-toggle-accessible-text-key-value:{sys:{id:lang-toggle-accessible-text-key-value,space:{},contentType:{id:keyValue}},fields:{slug:{en-CA:lang-toggle-accessible-slug},value:{en-CA:accesible toggle English,fr-CA:accessible toggle French}}},rich-text-key-value:{sys:{id:rich-text-key-value,space:{},contentType:{id:keyRichText}},fields:{slug:{en-CA:rich-text-key-slug},richText:{en-CA:{nodeType:document,data:{},content:[{nodeType:paragraph,data:{},content:[{nodeType:text,data:{},value:some text value}]}]}}}},some-complex-block-1:{sys:{id:some-complex-block-1,space:{},contentType:{id:complexBlock,type:Link,linkType:ContentType}},fields:{title:{en-CA:complex block title,fr-CA:complex block title - fr},body:{en-CA:{nodeType:document,data:{},content:[{nodeType:paragraph,data:{},content:[{nodeType:text,data:{},value:some paragraph in the complex block}]}]}}}},some-complex-block-2:{sys:{id:some-complex-block-2,space:{},contentType:{id:complexBlock,type:Link,linkType:ContentType}},fields:{title:{en-CA:complex block 2 title,fr-CA:complex block 2 title - fr},body:{en-CA:{nodeType:document,data:{},content:[{nodeType:paragraph,data:{},content:[{nodeType:text,data:{},value:some paragraph in the complex block 2}]}]}}}},some-block-group:{sys:{id:some-block-group,space:{},contentType:{id:modularLayout}},fields:{blocks:{en-CA:[{sys:{type:Link,linkType:Entry,id:some-complex-block-1}},{sys:{type:Link,linkType:Entry,id:some-complex-block-2}}]}}}}
      </div>
      <h2>
        linkedAssets
      </h2>
      <div>
        {test-img.png:{sys:{id:test-img.png},fields:{title:{en-CA:some title,fr-CA:some title fr},description:{en-CA:some description,fr-CA:some description fr},file:{en-CA:{url:/test-img.png,fileName:filename.png,contentType:image/png,details:{}}}}},test-img2.png:{sys:{id:test-img2.png},fields:{file:{en-CA:{url:/test-img2.png,fileName:filename2.png,contentType:image/png,details:{}}}}}}
      </div>
      <h2>
        keyValues
      </h2>
      <div>
        {lang-toggle-slug:{sys:{id:lang-toggle-key-value,space:{},contentType:{id:keyValue}},fields:{slug:{en-CA:lang-toggle-slug},value:{en-CA:toggle English,fr-CA:toggle French}}},rich-text-key-slug:{sys:{id:rich-text-key-value,space:{},contentType:{id:keyRichText}},fields:{slug:{en-CA:rich-text-key-slug},richText:{en-CA:{nodeType:document,data:{},content:[{nodeType:paragraph,data:{},content:[{nodeType:text,data:{},value:some text value}]}]}}}}}
      </div>
      <h2>
        PathMap
      </h2>
      <div>
        {/:{page:/,query:{lang:en-CA,pageId:site-home-page}},/home-pg-fr:{page:/,query:{lang:fr-CA,pageId:site-home-page}},/about-pg-en:{page:/,query:{lang:en-CA,pageId:about-us-page}},/about-pg-fr:{page:/,query:{lang:fr-CA,pageId:about-us-page}},/refreshed-page:{page:/,query:{lang:en-CA,pageId:REFRESHED-site-home-page}}}
      </div>
      <h2>
        lang
      </h2>
      <div>
        en-CA
      </div>
      <h2>
        previewAssets
      </h2>
      <div>
        {test-img.png:{url:data:image/png;base64,someUriData,meta:{width:10,height:10}},test-img2.png:{url:data:image/png;base64,someUriData2,meta:{width:10,height:10}}}
      </div>
      <h2>
        Render Link using fieldOrFallback
      </h2>
      {sys:{id:about-us-page-link,space:{},contentType:{sys:{type:Link,linkType:Entry,id:pageLink}}},fields:{title:{en-CA:page link title,fr-CA:page link title fr},page:{en-CA:{sys:{id:about-us-page},fields:{slug:{en-CA:some slug}}}},hash:{en-CA:s}}}
      <h2>
        langSwitchLink
      </h2>
      <div>
        {href:/home-pg-fr,text:toggle English,accessibleText:accesible toggle English}
      </div>
    </div>
  </MockChildComponent>
</SiteData>"
`;
